src/test/scala/chiselTests/LayerSpec.scala

```diff
@@ -45,6 +46,26 @@ class LayerSpec extends ChiselFlatSpec with Utils with MatchesAndOmits {
     )()
   }
 
+  it should "allow for defines in a layer to drive layer-colored probes" in {
+
+    class Foo extends RawModule {
+      val in = IO(Input(Bool()))
+      val a = IO(Output(Probe(Bool(), A)))
+      val b = IO(Output(Probe(Bool(), A.B)))
+      layer.block(A) {
+        define(a, ProbeValue(in))
+        layer.block(A.B) {
+          define(b, ProbeValue(in))
+        }
+      }
+    }
+
+    matchesAndOmits(ChiselStage.emitCHIRRTL(new Foo))(
+      "define a = probe(in)",
+      "define b = probe(in)"
+    )()
+  }
```

Comment:
Can we have a negative test of what happens if you try to drive a layer-colored probe from the wrong layer or from no layer?

Reply:
There's currently no checking of this. `firtool` will error about it if you do this. It's probably better to duplicate the check of FIRRTL semantics in Chisel here. I'll see about adding this.

Reply:
This logic was recently tweaked a bit, but here's the basic place where we check lexical scoping: https://github.com/chipsalliance/chisel/blob/4ffbd99e05fc7832466bb32389a81653862b2ffe/core/src/main/scala/chisel3/Data.scala#L585

Reply:
I added this to the checks of the legality of a define (and tests showing it works). This is included in https://github.com/chipsalliance/chisel/pull/3744/commits/bb5d02cafb27b496bd23d584493eb5a0660363fc. Test is added in https://github.com/chipsalliance/chisel/pull/3744/commits/f83f4b61b6d104c23a46392359a32431cd467597.